File: chatgpt_terminal.py
from openai import OpenAI

client = OpenAI(api_key = "api_key")

def chat_gpt(prompt):
    try:
        response = client.chat.completions.create(
            model="gpt-3.5-turbo",
            messages=[{"role": "user", "content": prompt}]
        )

        return response.choices[0].message.content.strip()
    except Exception as e:
        return f"Error: {str(e)}"

if __name__ == "__main__":
    print("Welcome to ChatGPT CLI! Type 'exit' to quit.")
    while True:
        user_input = input("You: ")
        if user_input.lower() == 'exit':
            break

        response = chat_gpt(user_input)
        print(f"ChatGPT: {response}")

File: chatgpt_terminal.ts
import { ClientOptions, OpenAI } from "openai";
import * as readlineSync from "readline-sync";

// Replace with your actual API key
const apiKey =
  "api_key";

// Initialize the OpenAI client
const client = new OpenAI({
  apiKey: apiKey,
} as ClientOptions);

async function chatGpt(prompt: string): Promise<string> {
  try {
    // Call OpenAI API (using the chat completion endpoint)
    const response = await client.chat.completions.create({
      model: "gpt-3.5-turbo",
      messages: [{ role: "user", content: prompt }],
    });

    const choice = response.choices[0];

    if (choice && choice.message && choice.message.content) {
        return choice.message.content.trim();
    } else {
        return "No response";
    }
  } catch (error) {
    return `Error: ${(error as Error).message}`;
  }
}

async function main() {
  console.log("Welcome to ChatGPT CLI! Type 'exit' to quit.");

  while (true) {
    const userInput = readlineSync.question("You: ");
    if (userInput.toLowerCase() === "exit") {
      break;
    }

    const response = await chatGpt(userInput);
    console.log(`ChatGPT: ${response}`);
  }
}

main().catch(console.error);
